### DISCUSSION TITLE
Development of a Connector for ReductStore as a Data Sink for MQTT

### DISCUSSION THREAD

**Discussion by AnthonyCvn** [emqx/emqx#12900] [Ideas]
Hello everyone,

I am planning to develop a connector that integrates EMQX with [ReductStore](https://www.reduct.store/), a time-series object store optimized for edge computing. The goal is to enable ReductStore to serve as a data sink for MQTT messages.

I am looking for feedback, suggestions, and any potential interest in collaboration. Does anyone have experience with similar integrations, or insights on the best practices for developing such a connector?

Thank you!

**Comment by zmstone** [comment]
Just had a quick kook, there seems to be a http api for dada ingestion?
https://www.reduct.store/docs/http-api/entry-api

Maybe webhook can work?

**Reply by AnthonyCvn** [reply]
Webhook connector and passing data to ReductStore API? That might work, but I was thinking about building a full connector integration like MongoDB: https://www.emqx.io/docs/en/latest/data-integration/data-bridge-mongodb.html

**Reply by AnthonyCvn** [reply]
@zmstone do you think we can develop such integration for ReductStore? I want to make sure you would accept us on emqx repo before we go through the whole development.

**Reply by zmstone** [reply]
Hi @AnthonyCvn 
Contributions are welcome!

You can maybe start from developing it as a plugin. Follow the instructions here: https://github.com/emqx/emqx-plugin-template
We are adding UI support for plugins from 5.7 (WIP, please wait for the repo update to see examples)

To add it as a native integration, it's a bit more complicated:

- Since this is outside our of roadmap, you may not expect us to be very responsive when it comes to support your development.
- After the bridge design refactoring (v1 -> v2), I have to say it's not quite straightforward to create a new bridge, places need to be changed are a bit scattered, and there is no formal doc to describe the HOW-TOs (we will prioritise this work after the bridge v1 deprecation).
- I18n for doc-generation and swagger API spec are hosted externally here: https://github.com/emqx/emqx-i18n, unsure how much time we'll need to allocate to support the translation.
- There would be a need to integrate it with the dashboard UI which is a frontend repo hosted here: https://github.com/emqx/emqx-dashboard5
  - There had not been any UI contribution from external so far, so the collaboration might not be as smooth there if you want to contribute
  - We'll need to do some UX demo before accepting the design.
  - There is also i18n part of the UI.

There are maybe other things which I cannot recall from top of my head. but I think we'll figure that out along the way.

**Reply by AnthonyCvn** [reply]
Hi @zmstone,

Thanks for the details. I appreciate the opportunity to contribute and understand the complexities involved. Excited to work on this!
